platform/platform_tests/distance_tests.cpp

```diff
@@ -346,4 +366,34 @@ UNIT_TEST(Distance_FormattedDistance)
   }
 }
 
+UNIT_TEST(Distance_ToStringPrecisionLocale)
```

Comment:
1. Let's move it to the related tests file. I guess, it's `measurement_tests.cpp`
2. I don't really like this solution.
   You can mock locale as it's done for display units with `ScopedSettings guard`.
   Create `ScopedLocale guard` class, pass separators to the constructor and overwrite current locale settings. In destructor - restore locale.
   Check `UNIT_TEST(Distance_CreateFormatted)` as an example

This solution won't require installing additional locales to the system.

Reply:
Testing specific locales in a controlled environment is ok. You can edit our .github/workflows actions by adding `sudo locale-gen ru_RU.UTF-8` commands in the beginning.

Reply:
I've added the `sudo locale-gen xx_YY.UTF-8` commands to `.github/workflows/linux_check.yaml` for different languages (en, es, fr & ru) and now the linux checks seem to work Ok.

Anyhow, the macOS checks still do not work. Instead of the linux command `locale-gen`, in macOS it looks like the command to use is `localedef`, but I still cannot get it working. I will make further investigations in the incoming days, but any help is welcomed.

Reply:
That will definetly break tests on local machines without all required locales.
At least, consider covering such test cases with `#if CI`
```cpp
    TestData testData[] = {
          { "en_US.UTF-8",     "9.8",      "12,345"},
#if CI
          { "es_ES.UTF-8",     "9,8",      "12.345"},
          { "fr_FR.UTF-8",     "9,8",      "12 345"},
          { "ru_RU.UTF-8",     "9,8",      "12 345"}
#endif
    };
```

We can pass `CI` var through cmake when it's defined in the environment. CI services like GitHub Actions always have this environment variable.

Reply:
Let me describe the latest tests and implementations that I've perfomed today:

- Finally it was not needed to use the command `localedef` in macOS. I only had to rename the locales from `xx_YY.UTF8` to `xx_YY.UTF-8` to get them working in macOS.
- I found out that in macOS the call to `std::locale::global()` does not affect the behavior of class `NSLocale`, so I've had to implement a new call `GetLocale(std::string)` in `locale.mm` order to obtain a new Locale struct based on the locale name in macOS.
- Now comes the funny part: I've made tests with 4 different locales: EN, ES, FR and RU. With EN and ES there are no problems, but with FR locale the grouping separator character in Linux is a normal whitespace (\u0020), but in macOS it's a Narrow Non-Breaking Space (NNBSP, \u202F). And with RU locale, in Linux it's also a normal whitespace (\u0020), but in macOS it's a standard Non-Breaking Space (NBSP, \u00A0).

There's a mix of many different types of space characters in some locales and in different platforms, and the only possible work-around is to fill the testing code with `#ifdef` for the different platforms and locales.

Taking into consideration all these tests and the comment from @AndrewShkrob...

> That will definetly break tests on local machines without all required locales.
> At least, consider covering such test cases with #if CI

... I think that it's no worth the implementation of the UTs for the `ToStringPrecision()` function with different locales.

Let me know your opinion whether we shall continue to implement these UTs.

Reply:
Finally I decided to remove the UTs for the `ToStringPrecision()` function with different locales.

Reply:
Testing it on different OSes with different locales actually was quite useful )

Reply:
Ok, I've brought back the UTs for the `ToStringPrecision()` function with different locales.

As the grouping separator is different for Linux and MacOS in some languages, the `UNIT_TEST(ToStringPrecisionLocale)` contains `#ifdefs` for different test results in different OS's.

I've included also a `#if CI` as suggested by @AndrewShkrob, so these tests are only performed in the GitHub workflows, and local tests do not require locales to be installed.